public/modules/moduleMetadataService.js

```diff
@@ -1,8 +1,8 @@
 angular.module('newApp')
 .factory('moduleMetadata', function($http) {
     return {
-        get : function() {
-            return $http.get('/api/units');
+        get : function(moduleID) {
+            return $http.get('/api/units' + (moduleID ? '/' + moduleID : ''));
```

Comment:
In my opinion, it's worthwhile having separate methods for these behaviors, such as `getAll()` and `get()`.